Add Better Region Example to SetCountryCode

michael-hawker opened the pull request:
This is to update the UWP doc with the proper guidance on setting the country code for the user based on discussion in issue #54.

This is the recommendation made in the [official sample](https://github.com/Microsoft/AppCenter-SDK-DotNet/blob/master/Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.UWP/App.xaml.cs) as well.

michael-hawker committed:
Add Better Region Example to SetCountryCode

This is to update the UWP doc with the proper guidance on setting the country code for the user.

This is the recommendation made in the [official sample](https://github.com/Microsoft/AppCenter-SDK-DotNet/blob/master/Apps/Contoso.Forms.Demo/Contoso.Forms.Demo.UWP/App.xaml.cs) as well.

iamclement reviewed (changes requested):
Thanks a lot for the PR. Though the `Windows.Globalization.GeographicRegion().CodeTwoLetter` doesn't exactly fit the SDK use case. The SDK reports this country code as carrier country code which means it's not supposed to be provided by the user. There was a discussion around this in the [App Center DotNet GH repository](https://github.com/Microsoft/AppCenter-SDK-DotNet/issues/362). We would need a sample using the location service I guess. I'll create a work item on our side to update the documentation with a location service sample but not sure when this will be picked up. So it's up to you @michael-hawker if you want to take it in such case you can do it in this PR or we close the PR and we take it on our side?

michael-hawker committed:
More detail to clarify the SetCountryCode example

Clarify that the Globalization call uses a user provided value and not an actual location.

michael-hawker commented:
@clpolet glad to know there's been discussion, but the linked issue is pretty old still.  I pinged it for an update and to resume the discussion.

In the meantime, I've clarified the example, but had to leave it a bit open as well as I don't know the proper solution.  Let me know how that looks to you.

However, considering that this is currently the usage in the sample, it'd be good to have this matching in the docs, as I didn't know there was a reference sample and just pulled info from the docs when implementing my app.

iamclement commented:
@michael-hawker you did right, it did resume the topic :slightly_smiling_face:. We are taking it from here. We are working on a code snippet using the location service. Both documentation and test apps will be updated with the new snippet. Stay tuned.

jwargo commented:
@michael-hawker closing this PR as @clpolet will address this in a different stream.
